README.md

```diff
@@ -200,6 +203,19 @@ Will run django tests using `--keepdb` and `--parallel`. You may pass a path to
 
 To add a new **backend** dependency, run `poetry add {dependency}`. If the dependency should be only available for development user append `-G dev` to the command.
 
+### API Schema
+
+We use the DRF-Spectacular tool to generate an OpenAPI schema from our Django Rest Framework API. The OpenAPI schema serves as the backbone for generating client code, creating comprehensive API documentation, and more.
+
+The API documentation pages are accessible at `http://localhost:8000/api/schema/swagger-ui/` or `http://localhost:8000/api/schema/redoc/`.
+
+> [!IMPORTANT]
+> Anytime a view is created, updated, or removed, the schema must be updated to reflect the changes. Failing to do so can lead to outdated client code or documentation.
+> 
+> To update the schema, run:
+> - If you are using Docker: `make docker_backend_update_schema`
+> - If you are not using Docker: `poetry run python manage.py spectacular --color --file schema.yml`
```

Comment:
I haven't implemented a mechanism to automatically detect view changes and update the schema. Maybe this can be considered later?

Reply:
Yes, please track in a new issue.
Also, I think you have to call spectacular somewhere in the deploy process?

Reply:
> Also, I think you have to call spectacular somewhere in the deploy process?
> Or the yml stays in the git? Because if latter, there's no need to integrate with deploy process.

The `schema.yml` currently stays in git.

---

> One thing we could do is detect in pre-commit if the schema.yml is outdated.

Cool, I added a new pre-commit hook to always run the generate command when there is a staged file from `backend/` (commit https://github.com/vintasoftware/django-react-boilerplate/pull/653/commits/f8fa673f02658fa6300657c9036e8f128a327326).

Reply:
Nice, that solves it.